rename VSCE, VSCE API, SDK, and CLI

davenice opened the pull request:
**What It Does**
#403 

**How to Test**
<!-- If a bug has been fixed, how can reviewers verify that the change(s) fixed it? -->

**Review Checklist**
I certify that I have:
- [x] tested my changes
- [ ] added/updated automated tests
- [x] updated the changelog
- [x] considered whether the docs need updating
- [x] followed the [contribution guidelines](https://github.com/zowe/zowe-cli/blob/master/CONTRIBUTING.md)

**Additional Comments**
<!-- Anything else noteworthy about this pull request. This section is optional. -->

adam-wolfe commented on packages/cli/README.md:
```diff
@@ -2,7 +2,7 @@
 
```
Would also update the heading in line 1 of this file

davenice replied:
Great spot! Thank you. Fixed the SDK one too.

AndrewTwydell commented on packages/sdk/README.md:
The SDK package README has the CLI plugin title 🤔 Perhaps that should be updated!

AndrewTwydell reviewed (approved):
Other than Adam's comment about line 1 of the CLI README and my comment on the SDK README, I think it looks good 👍🏼

davenice commented:
OK - changes made. Happy to approve, @adam-wolfe ?

jt-nti commented on packages/cli/README.md:
```diff
@@ -1,8 +1,8 @@
-# IBM® CICS® Plug-in for Zowe CLI
+# Zowe&reg; CLI Plug-in for IBM&reg; CICS&reg; Transaction Server
```
Trade mark in title

jt-nti commented on packages/sdk/README.md:
```diff
@@ -1,8 +1,8 @@
-# IBM® CICS® Plug-in for Zowe CLI
+# Zowe&reg; CLI Plug-in for IBM&reg; CICS&reg; Transaction Server
```
Trademark in title

jt-nti commented on packages/vsce-api/README.md:
```diff
@@ -1,12 +1,12 @@
-# IBM CICS for Zowe Explorer API
+# Zowe&reg; Explorer for IBM&reg; CICS&reg; Transaction Server API
```
Trademark in title

jt-nti commented on packages/vsce/docs/Troubleshooting.md:
```diff
@@ -1,6 +1,6 @@
-# IBM CICS for Zowe Explorer extension Troubleshooting
+# Zowe Explorer for IBM CICS Transaction Server extension Troubleshooting
```
No trademark in title

jt-nti commented on packages/vsce/docs/Uninstallation.md:
```diff
@@ -1,17 +1,17 @@
-# IBM CICS for Zowe Explorer Extension Uninstallation
+# Zowe Explorer for IBM CICS Transaction Server Extension Uninstallation
```
No trademark in title

jt-nti commented on packages/vsce/docs/installation-guide.md:
```diff
@@ -1,4 +1,4 @@
-# IBM CICS for Zowe Explorer Extension Installation
+# Zowe Explorer for IBM CICS Transaction Server Extension Installation
```
No trademark in title

jt-nti reviewed (changes requested):
I don't think the [Zowe documentation style guide](https://docs.zowe.org/stable/contribute/contributing/#documentation-style-guide) mentions trademarks for IBM docs we don't put trademarks in titles. There's a mix in this PR which I've hopefully added comments for.
Depending on what @anaxceron recommends, it might be better to take the trademarks out of the titles and add them to the first non-title occurrence instead?

adam-wolfe commented:
> > I don't think the [Zowe documentation style guide](https://docs.zowe.org/stable/contribute/contributing/#documentation-style-guide) mentions trademarks for IBM docs we don't put trademarks in titles. There's a mix in this PR which I've hopefully added comments for. Depending on what @anaxceron recommends, it might be better to take the trademarks out of the titles and add them to the first non-title occurrence instead?
> 
> @jt-nti @davenice -- Then let's use this as an opportunity to make a Zowe style standard for trademarks (and similar symbols). Let's avoid symbols on titles and use them on first non-title reference, as James said. I'll make a PR to add this to the style guide.

This sounds fine to me, but I will check with the Linux Foundation to confirm from their standpoint.

davenice commented:
I've moved the trademark symbols to be the first use after the title.
I've added "previously named" for the CLI and VSCE readmes to help with search and understanding.
I've removed most of the content of the SDK readme because it was the CLI readme.

I'm happy to move the trademark symbols back - in a subsequent PR if necessary - if Adam's research suggests Zowe adopt that policy.

davenice commented:
@jt-nti / @anaxceron - a quick re-review would be welcome, please!

anaxceron commented on packages/cli/README.md:
```diff
@@ -1,8 +1,8 @@
 [![codecov](https://codecov.io/gh/zowe/zowe-cli-cics-plugin/branch/master/graph/badge.svg)](https://codecov.io/gh/zowe/zowe-cli-cics-plugin)
 
-The IBM CICS Plug-in for Zowe CLI lets you extend Zowe CLI to interact with IBM CICS programs and transactions.
+The Zowe&reg; CLI Plug-in for IBM&reg; CICS&reg; Transaction Server (previously named IBM CICS Plug-in for Zowe CLI) lets you extend Zowe CLI to interact with IBM CICS programs and transactions.
```
```suggestion
The Zowe&reg; CLI Plug-in for IBM&reg; CICS&reg; Transaction Server (formerly IBM CICS Plug-in for Zowe CLI) lets you extend Zowe CLI to interact with IBM CICS programs and transactions.
```

anaxceron commented on packages/vsce/README.md:
```diff
@@ -1,15 +1,15 @@
 [![codecov](https://codecov.io/gh/zowe/cics-for-zowe-client/branch/main/graph/badge.svg)](https://app.codecov.io/gh/zowe/cics-for-zowe-client/branch/main/)
 
-This CICS Extension for Zowe Explorer adds additional functionality to the popular VS Code extension, [Zowe Explorer](https://github.com/zowe/vscode-extension-for-zowe). This extension allows interactions with CICS regions and resources, and the ability to run commands against them.
+This Zowe&reg; Explorer for IBM&reg; CICS&reg; Transaction Server extension (previously named IBM CICS for Zowe Explorer) adds additional functionality to the popular VS Code extension, [Zowe Explorer](https://github.com/zowe/vscode-extension-for-zowe). This extension allows interactions with CICS regions and resources, and the ability to run commands against them.
```
```suggestion
This Zowe&reg; Explorer for IBM&reg; CICS&reg; Transaction Server extension (formerly IBM CICS for Zowe Explorer) adds additional functionality to the popular VS Code extension, [Zowe Explorer](https://github.com/zowe/vscode-extension-for-zowe). This extension allows interactions with CICS regions and resources, and the ability to run commands against them.
```

adam-wolfe replied:
I would say "**The** Zowe&reg; Explorer for IBM&reg;"...

anaxceron reviewed (approved):
Thank you for taking care of this @davenice!

adam-wolfe commented:
Tagging @Joe-Winchester 

During TSC there were also concerns about including trademark symbol in URLs.

adam-wolfe commented:
We have agreed to remove the trademark symbols from the titles.

davenice commented on packages/vsce/README.md:
```diff
@@ -1,15 +1,15 @@
 [![codecov](https://codecov.io/gh/zowe/cics-for-zowe-client/branch/main/graph/badge.svg)](https://app.codecov.io/gh/zowe/cics-for-zowe-client/branch/main/)
 
-This CICS Extension for Zowe Explorer adds additional functionality to the popular VS Code extension, [Zowe Explorer](https://github.com/zowe/vscode-extension-for-zowe). This extension allows interactions with CICS regions and resources, and the ability to run commands against them.
+The Zowe&reg; Explorer for IBM&reg; CICS&reg; Transaction Server extension (previously named IBM CICS for Zowe Explorer) adds additional functionality to the popular VS Code extension, [Zowe Explorer](https://github.com/zowe/vscode-extension-for-zowe). This extension allows interactions with CICS regions and resources, and the ability to run commands against them.
```
```suggestion
The Zowe&reg; Explorer for IBM&reg; CICS&reg; Transaction Server extension (formerly named IBM CICS for Zowe Explorer) adds additional functionality to the popular VS Code extension, [Zowe Explorer](https://github.com/zowe/vscode-extension-for-zowe). This extension allows interactions with CICS regions and resources, and the ability to run commands against them.
```

davenice commented:
Right. I just found @anaxceron's suggestion around "formerly" - thanks.

@jt-nti - in the situation with troubleshooting, what would we do in IBM doc? Let's take the same stance here.

jt-nti commented:
@davenice troubleshooting was just an example- contributing is the same I think, and there may be others. As far as I know the idea is to show the trademark for the first use in normal text, i.e. not titles, or things like `zowe config list` examples.

davenice commented:
@jt-nti I've updated other guides. I fixed some clearly out of date information. If you want to reword, please make a suggestion I can accept to minimise the amount of times we go round the loop on this! :-D

Joe-Winchester reviewed (approved):
These all look excellent to me !
